store/store.go

```diff
@@ -531,11 +531,57 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+// RefreshCandidate contains information for the store about the currently
+// installed snap so that the store can decide what update we should see
+type RefreshCandidate struct {
+	SnapID   string
+	Revision snap.Revision
+	Epoch    string
+	DevMode  bool
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+	// the desired channel
+	Channel string
+}
+
+// the exact bits that we need to send to the store
+type currentSnapJson struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision,omitempty"`
+	Epoch    string `json:"epoch"`
+	DevMode  bool   `json:"devmode"`
```

Comment:
I do understand you need to let the store know the confinement switch was used when the snap was installed, and that it could be different from the confinement reported by the revision of that snap in the store, but the client could still send the `confinement` parameter with the actual (installation) value, correct?

We agree the store needs to know how the revision was actually installed to decide if there is/which update to return. My concern is that if eventually there is a new possible confinement value, having a boolean flag doesn't seem enough, meanwhile having the confinement field will always report the confinement the revision has been installed with, and the store will have the information to decide what to do accordingly. Also, at least from the store perspective, 'devmode' is a particular value for confinement so it looks odd having it as a key.